.github/workflows/release.yml

```diff
@@ -98,18 +98,14 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       packages: write
-    needs: build-artifact
+    needs: 
+      - build-artifact
+      - pypi-publish
+    if: |
+      always() &&
+      needs.build-artifact.result == 'success' &&
+      (needs.pypi-publish.result == 'success' || needs.pypi-publish.result == 'skipped')  
```

Comment:
why the check for pypi-publish here ?
dont we always want to grab the release from the prebuilt wheel?

Reply:
I assumed that we wanted to build from the pypi published artifact IF we actually publish it, if not, then I can just remove that bit.

Reply:
I think it makes sense to also put the 'publish-to-pypi' job as a pre-requisite for the docker build job for stable releases. That way we can ensure that the release artifact is actually published to pypi before we build a stable Docker image.

Reply:
Also think we should add an additional check to the jobs for bumping the add-on / creating the release notes. They should not run for dev builds.

Reply:
That would be a tricky condition to write though.

`if: |
      always() &&
      needs.build-artifact.result == 'success' &&
      ((needs.pypi-publish.result == 'success' && GitHub.events.release  == 'true') || (needs.pypi-publish.result ==     'skipped' && GitHub.events.release  == 'false'))`

Or something weird like this

Reply:
There's no such thing as weird code if there's a proper comment that goes along with it ;-)

Reply:
I prefer to avoid code that needs to be commented to explain it, but you're right, waiting for PYPI makes sense for release builds, and a comment at least serves to explain in case of confusion.

Reply:
yeah - its okay if we dont install FROM pypi (as that adds an annoying delay) but we must ensure that for stable builds, the pypi task did complete